share/functions/__fish_git_prompt.fish

```diff
@@ -471,14 +471,20 @@ end
 set -g ___fish_git_prompt_status_order stagedstate invalidstate dirtystate untrackedfiles
 
 function __fish_git_prompt_informative_status
-
     set -l changedFiles (command git diff --name-status | cut -c 1-2)
     set -l stagedFiles (command git diff --staged --name-status | cut -c 1-2)
+    set -l untrackedfiles 0
 
     set -l dirtystate (math (count $changedFiles) - (count (echo $changedFiles | grep "U")))
     set -l invalidstate (count (echo $stagedFiles | grep "U"))
     set -l stagedstate (math (count $stagedFiles) - $invalidstate)
-    set -l untrackedfiles (count (command git ls-files --others --exclude-standard))
+
+    # Untracked files may cause slow prompt generation.
+    # For this reason, it should be possible to avoid counting the files
+    # by setting __fish_git_prompt_showuntrackedfiles explicitly to "false".
+    if test "$__fish_git_prompt_showuntrackedfiles" != false
```

Comment:
@zanchey and @faho, What do you think? I'm inclined to make the behavior consistent rather than add yet another subtle behavior that is incompatible with how the var is treated in `__fish_git_prompt`. Especially with a comment to that effect buried deep in the script rather than at the top where this var is documented.